src/core/obj.js

```diff
@@ -177,6 +177,48 @@ class Catalog {
     return (root.items.length > 0 ? root.items : null);
   }
 
+  get permissions() {
+    let permissions = null;
+    try {
+      permissions = this._readPermissions();
+    } catch (ex) {
+      if (ex instanceof MissingDataException) {
+        throw ex;
+      }
+      warn('Unable to read permissions.');
+    }
+    return shadow(this, 'permissions', permissions);
+  }
+
+  /**
+   * @private
+   */
+  _readPermissions() {
+    const encrypt = this.xref.trailer.get('Encrypt');
+    if (!isDict(encrypt)) {
+      return null;
+    }
+
+    let flags = encrypt.get('P');
+    if (!isNum(flags)) {
+      return null;
+    }
+
+    // PDF integer objects are represented internally in signed 2's complement
+    // form. Therefore, convert the signed decimal integer to a signed 2's
+    // complement binary integer so we can use regular bitwise operations on it.
+    flags += 2 ** 32;
```

Comment:
I assuming that you've already checked, but otherwise please make sure that Webpack/Babel is able to convert this into code compatible with older browsers. (Note that I'm expecting this to "just work" given [this option](https://github.com/mozilla/pdf.js/blob/master/external/builder/preprocessor2.js#L291)).

Reply:
Good point. I found that `babel-present-env` fortunately includes the exponentiation operator plugin (see https://github.com/babel/babel/blob/master/packages/babel-preset-env/package.json#L36), so that should be fine.